### PR TITLE
VSTW-2331: Add ability to set timeouts for clients

### PR DESCRIPTION
**Pull request by base10**
* Removes protected method test
  * In Ruby 2.0 and later, protected methods will no longer answer true
    to respond_to? questions.
* Addresses RSpec 2.99 deprecation warnings
* Addresses test structure, word separation
* Makes the open and read timeouts configurable
* Moves the default timeout value to a constant

### CODE REVIEW

**Comment by base10** [comment]
Ref. only PR. Will need to open a separate PR with the upstream project.

**Comment by base10** [comment]
See also: https://github.com/vitalsource/stargate/pull/1577

File: lib/marketingcloudsdk/client.rb [hunk 1]
```diff
@@ -3,46 +3,46 @@
...
 
-WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE 
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 
 USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-=end 
-
+=end
+require 'pry'
```
Comment (peburrows) [inline lib/marketingcloudsdk/client.rb]: Probably don't want to `require 'pry'` here since you're only including it in the development bundle.

Comment (base10) [reply lib/marketingcloudsdk/client.rb]: Good call.

**Review by victor-seo — APPROVED** [review]

**Review by jcbmartens — APPROVED** [review]

**Comment by base10** [comment]
No longer using ExactTarget 🗑️

**Pull request closed without merging**